vcd/api_test.go

```diff
@@ -72,3 +84,77 @@ func TestTags(t *testing.T) {
 		showTags()
 	}
 }
+
+// Finds the current directory, through the path of this running test
+func getCurrentDir() string {
+	_, currentFilename, _, _ := runtime.Caller(0)
+	return filepath.Dir(currentFilename)
+}
+
+// Reads the version from the VERSION file in the root directory
+func getMajorVersion() string {
+
+	versionFile := path.Join(getCurrentDir(), "..", "VERSION")
+
+	// Checks whether the VERSION file exists
+	_, err := os.Stat(versionFile)
+	if os.IsNotExist(err) {
+		panic("Could not find VERSION file")
+	}
+
+	// Reads the version from the file
+	versionText, err := ioutil.ReadFile(versionFile)
+	if err != nil {
+		panic(fmt.Errorf("could not read VERSION file %s: %v", versionFile, err))
+	}
+
+	// The version is expected to be in the format v#.#.#
+	// We only need the first two numbers
+	reVersion := regexp.MustCompile(`v(\d+\.\d+)\.\d+`)
+	versionList := reVersion.FindAllStringSubmatch(string(versionText), -1)
+	if versionList == nil || len(versionList) == 0 {
+		panic("empty or non-formatted version found in VERSION file")
+	}
+	if versionList[0] == nil || len(versionList[0]) < 2 {
+		panic("unable to extract major version from VERSION file")
+	}
+	// A successful match will look like
+	// [][]string{[]string{"v2.0.0", "2.0"}}
+	// Where the first element is the full text matched, and the second one is the first captured text
+	return versionList[0][1]
+}
+
+// Checks that the provider header in index.html.markdown
+// has the version defined in the VERSION file
+func TestProviderVersion(t *testing.T) {
```

Comment:
I am thinking if this test should be guarded with a separate tag (means in different file).
Currently this kind of ruins your nice `go test -v .` message. I think I wouldn't want to see this test being run as no others are run at that time.

```
$  go test -v .
=== RUN   TestTags
--- FAIL: TestTags (0.00s)
    api_test.go:78: # No tags were defined
    api_test.go:53:
        # -----------------------------------------------------
        # Tags are required to run the tests
        # -----------------------------------------------------

        At least one of the following tags should be defined:

           * ALL :       Runs all the tests
           * functional: Runs all the acceptance tests
           * unit:       Runs unit tests that don't need a live vCD (currently unused, but we plan to)

           * catalog:    Runs catalog related tests (also catalog_item, media)
           * disk:       Runs disk related tests
           * network:    Runs network related tests
           * gateway:    Runs edge gateway related tests
           * org:        Runs org related tests
           * vapp:       Runs vapp related tests
           * vdc:        Runs vdc related tests
           * vm:         Runs vm related tests

        Examples:

        go test -tags functional -v -timeout=45m .
        go test -tags catalog -v -timeout=15m .
        go test -tags "org vdc" -v -timeout=5m .
=== RUN   TestProviderVersion
--- PASS: TestProviderVersion (0.00s)
    api_test.go:147: Found expected version <2.2> in index.html.markdown
FAIL
FAIL    github.com/terraform-providers/terraform-provider-vcd/v2/vcd    0.017s
```

Reply:
I did that on purpose. This check is so important that we want to run it regardless.

But if the team disagrees, I can either include this test in the `api` tag (it will run with `make test`) or create a new file and start using the `unit` tag, as it doesn't require a connection to run.

Reply:
I agree with @Didainius . And, yes, `unit` tag sounds good.

Reply:
Created the first unit test with `unit` tag.
Also available as `make testunit`

Running `make test` and `make testacc` will also run the unit test as a preliminary step